src/network_service/sending_targets_cache.rs

```diff
@@ -20,18 +20,21 @@ use std::{
 /// Time interval during which we attempt to resend a message to a given target before giving up.
 pub const RESEND_TIMEOUT: Duration = Duration::from_secs(30);
 
+struct Target {
+    // What state the target is at - sending or failed.
+    state: TargetState,
+    // Timestamp of the first send attempt to this target.
+    timestamp: Instant,
```

Comment:
If we try in a round robbin fashion, say 3 times then perhaps the time between each attempt should be at least 10 secs? So we keep cycling round until we have exhausted each attempt. I am wondering if there is a clever way we can arrange to code for a typical `send` method that can do these waits, but when we have multiple targets it wll not wat on each target ut cycle them all, but till not try to send until the min time between tries exhausts? 
However, if we try a 100 times in that period then it's also OK just as long as we have a min of 3 10 sec (say) timeouts between what we consider then `Failed attempt`. So each attempt may fail, but we have a timeout that says we could not send in TIMEOUT duration (and in that duration we have tried at least once)